spring-integration-core/src/main/java/org/springframework/integration/config/AggregatorFactoryBean.java

```diff
@@ -48,128 +49,267 @@
+		}
+		else {
+			if (this.methodName == null) {
+				outputProcessor = new MethodInvokingMessageGroupProcessor(this.processorBean);
+			}
+			else {
+				outputProcessor = new MethodInvokingMessageGroupProcessor(this.processorBean, this.methodName);
+			}
+		}
+		this.aggregator = new AggregatingMessageHandler(outputProcessor);
+
+		if (this.applicationContext != null) {
+			this.aggregator.setApplicationContext(this.applicationContext);
+		}
+
+		if (this.applicationEventPublisher != null) {
+			this.aggregator.setApplicationEventPublisher(applicationEventPublisher);
+		}
+
+		if (this.beanName != null) {
+			this.aggregator.setBeanName(this.beanName);
+		}
+
+		if (this.expireGroupsUponCompletion != null) {
+			this.aggregator.setExpireGroupsUponCompletion(this.expireGroupsUponCompletion);
+		}
+
+		if (this.sendTimeout != null) {
+			this.aggregator.setSendTimeout(this.sendTimeout);
+		}
+
+		if (this.outputChannelName != null) {
+			this.aggregator.setOutputChannelName(this.outputChannelName);
+		}
+
+		if (this.metrics != null) {
+			this.aggregator.configureMetrics(this.metrics);
+		}
+
+		if (this.channelResolver != null) {
+			this.aggregator.setChannelResolver(this.channelResolver);
+		}
+
+		if (this.statsEnabled != null) {
+			this.aggregator.setStatsEnabled(this.statsEnabled);
+		}
+
+		if (this.countsEnabled != null) {
+			this.aggregator.setCountsEnabled(this.countsEnabled);
+		}
+
+		if (this.lockRegistry != null) {
+			this.aggregator.setLockRegistry(this.lockRegistry);
+		}
+
+		if (this.messageStore != null) {
+			this.aggregator.setMessageStore(this.messageStore);
+		}
+
+		if (this.correlationStrategy != null) {
+			this.aggregator.setCorrelationStrategy(this.correlationStrategy);
+		}
+
+		if (this.releaseStrategy != null) {
+			this.aggregator.setReleaseStrategy(this.releaseStrategy);
+		}
+
+		if (this.groupTimeoutExpression != null) {
+			this.aggregator.setGroupTimeoutExpression(this.groupTimeoutExpression);
+		}
+
+		if (this.forceReleaseAdviceChain != null) {
+			this.aggregator.setForceReleaseAdviceChain(this.forceReleaseAdviceChain);
+		}
+
+		if (this.taskScheduler != null) {
+			this.aggregator.setTaskScheduler(this.taskScheduler);
+		}
+
+		if (this.discardChannel != null) {
+			this.aggregator.setDiscardChannel(this.discardChannel);
+		}
+
+		if (this.discardChannelName != null) {
+			this.aggregator.setDiscardChannelName(this.discardChannelName);
+		}
+
+		if (this.sendPartialResultOnExpiry != null) {
+			this.aggregator.setSendPartialResultOnExpiry(this.sendPartialResultOnExpiry);
+		}
+
+		if (this.minimumTimeoutForEmptyGroups != null) {
+			this.aggregator.setMinimumTimeoutForEmptyGroups(this.minimumTimeoutForEmptyGroups);
+		}
+
+		if (this.expireGroupsUponTimeout != null) {
+			this.aggregator.setExpireGroupsUponTimeout(this.expireGroupsUponTimeout);
+		}
+
 		return this.aggregator;
 	}
 
+	@Override
+	public Class<? extends MessageHandler> getObjectType() {
+		return AggregatingMessageHandler.class;
```

Comment:
OK. Why others don't do the same? `FilterFactoryBean`, `ServiceActivatorFactoryBean` etc.